distribution/lib/Standard/Base/0.0.0-dev/src/Meta.enso

```diff
@@ -217,6 +218,20 @@ type Polyglot
         lang_str = get_polyglot_language self.value
         if lang_str == "java" then Language.Java else Language.Unknown
 
+type Function
```

Comment:
There are many `type` in `Meta` - why don't we have a single `Meta` type and many constructors `Atom`, `Constructor`, `Primitive`, etc.? Is this just a heritage or is there a benefit of doing so?

Vote +1 for me to attempt to rewrite to single `Meta` type and constructors. Vote -1 for keeping status quo as much as possible. CCing @jdunkerley, @GregoryTravis, @radeusgd

Reply:
Yes, the common type is the problem. There is a [really long signature](https://github.com/enso-org/enso/blob/6249c79ffd35df73d3db0c0f869e88cedad327bc/distribution/lib/Standard/Base/0.0.0-dev/src/Meta.enso#L368) and I'd love to shorten it to `meta : Any -> Meta`.

> an just throw 'Illegal_State' for such wrong calls

Right now we are getting `No_Such_Method` errors. Rather than throwing the (or another) error, I'd just return some default, when sensible. If I rewrote to `Meta` with multiple constructors.